lib/src/carry_save_multiplier.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// carry_save_multiplier.dart
+// Implementation of pipeline multiplier module.
+//
+// 2023 February 21
+// Author: Max Korbel <max.korbel@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+
+/// A datatype class that hold result from [FullAdder].
+class FullAdderResult {
+  /// The output sum from full adder.
+  final sum = Logic(name: 'sum');
+
+  /// The output carry-out from full adder.
+  final cOut = Logic(name: 'c_out');
+}
+
+/// A [FullAdder] module that can perform simple adder. The full adder output
```

Comment:
Doc comment is a little confusing.  Let's describe it more specifically, something more like
```dart
/// A simple full-adder with inputs [a] and [b] to be added with a [carryIn].
```